File: chapter3_1/cpp/rainfall.cpp
#include "rainfall.h"

using namespace std;

sample::sample() {}
sample::sample(string d, double r)
: date(d)
, rainfall(r)
{}

double avg_rainfall(location &loc)
{
    double total = 0;
    for (const auto &sample : loc.samples)
    {
        total += sample.rainfall;
    }
    return total / loc.samples.size();
}

File: chapter3_1/rainfall.js
"use strict";
const addon = require('./cpp/build/Release/rainfall');

let location = {
    latitude: 40.71, longitude: -74.01,
    samples: [
        { date: "2016-06-07", rainfall: 2 },
        { date: "2016-08-12", rainfall: 0.5 }
    ]
};
let average = addon.avg_rainfall(location);

console.log("Average rain fall = " + average + " cm");
